### PR TITLE
[Operator Authoring] Memory efficient pointwise fusion

### PR DESCRIPTION
**Pull request by anijain2305**
@Chillee @bertmaher

### CODE REVIEW

File: functorch/_src/memory_efficient_op_authoring.py [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+    return compiled_function(fn, fw_compiler, bw_compiler, partition_fn)
+
+
+def tensorexpr_operator_authoring(fn, partition_fn):
+    fw_compiler = tensorexpr_compile
+    bw_compiler = tensorexpr_compile
+    return compiled_function(fn, fw_compiler, bw_compiler, partition_fn)
+
+
+def memory_efficient_operator_authoring(fn, compiler_name):
```
Comment (Chillee) [inline functorch/_src/memory_efficient_op_authoring.py:154]: provide a default here imo

File: functorch/_src/memory_efficient_op_authoring.py [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+    for i in range(0, 100):
+        attr = f"_tensor_constant{i}"
+        if hasattr(fx_module, attr):
+            setattr(fx_module, attr, getattr(fx_module, attr).to(inp_device))
+        else:
+            break
+
+    jit_module = torch.jit.trace(fx_module, flat_args)
+    jit_module = torch.jit.freeze(jit_module.eval())
+    torch._C._jit_trace_module(jit_module._c, tuple(flat_args))
```
Comment (Chillee) [inline functorch/_src/memory_efficient_op_authoring.py:113]: we need to wait for Nick's PR to be merged before this.

File: functorch/_src/memory_efficient_op_authoring.py [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+decomposition_rules = {}
+decomposition_rules[torch.ops.aten.tanh_backward] = tanh_backward_decomposition
+decomposition_rules[torch.ops.aten._s_where] = _s_where_decomposition
+decomposition_rules[torch.ops.aten.sigmoid_backward] = sigmoid_backward_decomposition
+decomposition_rules[torch.ops.aten.softplus_backward] = softplus_backward_decomposition
+decomposition_rules[torch.ops.aten.detach] = noop
+
+# TODO - Move to the python key based decomposition
+def decompose(fx_module):
```
Comment (Chillee) [inline functorch/_src/memory_efficient_op_authoring.py:37]: I wonder whether we should switch over to a "proper" decomposition implementation first.

Comment (Chillee) [reply functorch/_src/memory_efficient_op_authoring.py:37]: Eh... we can just do that later.

File: functorch/_src/memory_efficient_op_authoring.py (hunk 2 above)
Comment (anijain2305) [reply functorch/_src/memory_efficient_op_authoring.py:113]: Thats ok. I can disable the testing of TEKernel for now and push this code in.

**Pull request merged**